packages/x-charts/src/ChartsLabel/ChartsLabelMark.tsx

```diff
@@ -0,0 +1,172 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { styled } from '@mui/material/styles';
+import { ChartsLabelMarkClasses, labelMarkClasses, useUtilityClasses } from './labelMarkClasses';
+
+export interface ChartsLabelMarkProps {
+  /**
+   * Defines the max size of the mark.
+   *
+   * For the `line` type, the size is the length of the line.
+   * For all other types, the size is the width and height of the mark.
+   *
+   * @default type='square': 13
+   * @default type='line': 16
+   * @default type='circle': 15
+   */
+  // eslint-disable-next-line react/no-unused-prop-types
+  size?: number;
+  /**
+   * The type of the mark.
+   * @default 'square'
+   */
+  type?: 'square' | 'circle' | 'line';
```

Comment:
Would it be easier to pass directly the compoennt to render.

For example having a `labelMark` properties in the series that we defaultize in the `seriesFormatter` like we do with the colors. SUch that the info is also available from the tooltip

And we provide helper components `ChartSquareLabelMark`, `ChartCircleLabelMark`, `ChartLineLabelMark`

Reply:
Good point on the series formatter, I was starting to lean that way (ChartsLabel/ChartsLabelMark are reusable), your comment reinforces it 😃 

> And we provide helper components ChartSquareLabelMark, ChartCircleLabelMark, ChartLineLabelMark

Not sure about the advantages? How would you expect this to work?

```
series={[{
  data: [],
  labelMark: 'circle'
}]}
// or
series={[{
  data: [],
  labelMark: <ChartCircleLabelMark />
}]}
```

Reply:
More the second one. I think as soon as you open flexibility for those marks, the devs will want to add lots of variation.

And managing it with a limited number of string + a size params feels limited. So the proposal to directly pass a component.

Another option can be a `slots.markItem` which get the series config as a props.

So either each series defines its component, or a components that adapts according to the series. I sey "series" but for example the pie chart defines legend/tooltip marks per items.